### PR TITLE
Init and qa fixes

### PR DESCRIPTION
**Pull request by gandrewstone**
fix initialization ordering problem when txindex=1 and checklevel>=4
restore qa --tmppfx behavior  
Add quick-run capability to setwit_recovery test"

### CODE REVIEW

**Review by ptschip — APPROVED** [review]

File: qa/rpc-tests/test_framework/test_framework.py [hunk 1]
```diff
@@ -206,7 +206,7 @@ def main(self,argsOverride=None,bitcoinConfDict=None,wallets=None):
         logging.info("Random seed: %s" % self.randomseed)
 
         if self.options.tmpdir is None:
-            self.options.tmpdir = self.options.tmppfx
+            self.options.tmpdir = os.path.join(self.options.tmppfx, str(self.options.port_seed))
```
Comment (sickpig) [inline qa/rpc-tests/test_framework/test_framework.py]: why adding back seed port in the construction of `tmpdir` path?

File: qa/rpc-tests/segwit_recovery.py [hunk 2]
```diff
@@ -9,6 +9,8 @@
 """
 
 import time
+from test_framework.util import *
```
Comment (sickpig) [inline qa/rpc-tests/segwit_recovery.py]: why importing everything from `util`? I'd rather prefer adding what's actually needed to the `import` as stated at L41.

File: src/init.cpp [hunk 3]
```diff
@@ -1106,6 +1106,13 @@ bool AppInit2(Config &config, thread_group &threadGroup)
 
     bool fLoaded = false;
     StartTxAdmission(threadGroup);
+
+    if (fTxIndex)
+    {
```
Comment (sickpig) [inline src/init.cpp:1111]: what's this change about?

Comment (sickpig) [reply src/init.cpp:1111]: never mind, figured out!

File: qa/rpc-tests/test_framework/test_framework.py (hunk 1 above)
Comment (sickpig) [reply qa/rpc-tests/test_framework/test_framework.py]: I mean this is not needed to restore the `tmppfx` cause it was never removed in the first place.

Comment (sickpig) [reply qa/rpc-tests/test_framework/test_framework.py]: fixed in d639deb

**Review by sickpig — APPROVED** [review]

File: qa/rpc-tests/segwit_recovery.py (hunk 2 above)
Comment (gandrewstone) [reply qa/rpc-tests/segwit_recovery.py]: fixed

**Pull request merged**